Core/include/Acts/Seeding/CandidatesForMiddleSp.hpp

```diff
@@ -0,0 +1,187 @@
+  /// @param weight The quality of the triplet candidate
+  /// @param zOrigin The z-coordinate of the origin
+  /// @param isQuality Whether the triplet candidate is high or low quality
+  void push(std::vector<std::size_t>& storage, std::size_t& n,
+            const std::size_t& n_max, external_space_point_t& SpB,
+            external_space_point_t& SpM, external_space_point_t& SpT,
+            float weight, float zOrigin, bool isQuality);
+
+  /// @brief Check if an element exists in the collection. The element to be checked
+  /// is supposed to be in the n position of the collection.
+  /// @param n Index of the requested element
+  /// @param max_size Number of elements currently stored in the collection
+  /// @returns Whether the element exists
+  bool exists(const std::size_t& n, const std::size_t& max_size) const;
+
+  /// @brief Pop an element from a collection. The removal of the element from the collection
+  /// does not imply its destruction. In fact, the number of stored elements is
+  /// simply diminished by 1. The popped element is tecnically still available
+  /// at the end of the collection.
+  /// @param storage The collection
+  /// @param current_size The current number of element stored in the collection. The function will
+  /// diminish this value by 1
+  void pop(std::vector<std::size_t>& storage, std::size_t& current_size);
+
+  /// @brief Return the weight for a candidate
+  /// @param storage The collection in which the element is stored
+  /// @param n Index of the element in the collection
+  /// @returns The weight of the candidate
+  float weight(const std::vector<std::size_t>& storage, std::size_t n) const;
+
+  /// @brief Move an element up in the min heap tree. The function checks whether the element's
+  /// weight is lower of it's parent's weight. If so, it swaps them. Reiterate
+  /// the process untill the element is in the correct position on the tree
+  /// @param storage The collection
+  /// @param n The index of the element to place in the correct position
+  void bubbleup(std::vector<std::size_t>& storage, std::size_t n);
+
+  /// @brief Move an element down in the min heap tree. The function checks whether the elements's
+  /// weight is lower of it's child's weights. If so, it swaps the element with
+  /// the child with the lowest weight. Reiterate the process untill the element
+  /// is in the correct position on the tree
+  /// @param storage The collection
+  /// @param n The index of the element to place in the correct position
+  /// @param actual_size The current number of elements stored in the collection
+  void bubbledw(std::vector<std::size_t>& storage, std::size_t n,
+                std::size_t actual_size);
+
+  /// @brief Sdding a new triplet candidate to the collection. The function is called after the candidate has satisfied
+  /// all the selection criteria
+  /// @param storage The collection
+  /// @param n Current number of stored elements in the collection
```

Comment:
Isn't it known? I mean the current number?

Reply:
Same comment as above about the use of two different indices collections